plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -65,606 +68,1719 @@ describe('PersonState.update()', () => {
-                version: 0,
-            })
-        )
-        const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
-        expect(clickhousePersons.length).toEqual(1)
-        expect(clickhousePersons[0]).toEqual(
-            expect.objectContaining({
-                id: uuid.toString(),
-                properties: JSON.stringify({ foo: 'bar' }),
-                created_at: '2020-01-01 12:00:05.000',
-                version: 0,
-            })
-        )
-    })
+            // verify ClickHouse persons
+            const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
+            expect(clickhousePersons.length).toEqual(1)
+            expect(clickhousePersons[0]).toEqual(
+                expect.objectContaining({
+                    id: uuid.toString(),
+                    properties: JSON.stringify({ b: 4, c: 4, e: 4 }),
+                    created_at: timestampch,
+                    version: 1,
+                    is_identified: 0,
+                })
+            )
+
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['new-user']))
+
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+
+        it('updates is_identified', async () => {
+            await hub.db.createPerson(timestamp, {}, {}, {}, 2, null, false, uuid.toString(), ['new-user'])
 
-    it('merges people on $identify event', async () => {
-        await hub.db.createPerson(timestamp, { a: 1, b: 2 }, {}, {}, 2, null, false, uuid.toString(), ['old-user'])
-        await hub.db.createPerson(timestamp, { b: 3, c: 4 }, {}, {}, 2, null, false, uuid2.toString(), ['new-user'])
-
-        await personState({
-            event: '$identify',
-            distinct_id: 'new-user',
-            properties: {
-                $anon_distinct_id: 'old-user',
-            },
-        }).update()
-        await hub.db.kafkaProducer.flush()
-
-        const persons = await hub.db.fetchPersons()
-        expect(persons.length).toEqual(1)
-        expect(persons[0]).toEqual(
-            expect.objectContaining({
-                id: expect.any(Number),
-                uuid: uuid2.toString(),
-                properties: { a: 1, b: 3, c: 4 },
-                created_at: timestamp,
-                is_identified: true,
-                version: 1,
+            const state = personState({
+                event: '$pageview',
+                distinct_id: 'new-user',
             })
-        )
-        const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
-        expect(distinctIds).toEqual(expect.arrayContaining(['new-user', 'old-user']))
+            state.updateIsIdentified = true
```

Comment:
Let's avoid doing this.

The logic around `updateIsIdentified` is already really self-contained in the class so by mutating it here we're just creating a more confusing class.

Reply:
will remove, make sure we cover the case of only is_identified vs properties only updated